src/app/child-dev-project/children/children.service.ts

```diff
@@ -119,10 +120,15 @@ export class ChildrenService {
     );
   }
 
-  getNotesOf(entityId: string, noteProperty: string): Promise<Note[]> {
-    return this.dbIndexing.queryIndexDocs(
+  /**
+   * Query all notes that have been linked to the given other entity.
+   * @param entityId ID (with prefix!) of the related record
+   */
+  getNotesRelatedTo(entityId: string): Promise<Note[]> {
+    return this.dbIndexing.queryIndexDocsRange(
       Note,
-      `notes_index/by_${noteProperty}`,
+      `notes_related_index/note_by_relatedEntities`,
+      entityId + "", // in future, we can add a date limit here (see AttendanceService.getEventsForActivity)
```

Comment:
```suggestion
      entityId + ":", // in future, we can add a date limit here (see AttendanceService.getEventsForActivity)
```
Same for `endkey` so we don't class with same prefixes, e.g. `Child` and `ChildSchoolRelation` both match the prefix `Child`